### DISCUSSION TITLE
How to forbid to create and upload folders to file manager

### DISCUSSION THREAD

**Discussion by backpack-operations** [Laravel-Backpack/community-forum#671] [Q&A (Help)] [answered]
Received this by email today:

> Hey! I'm PHP developer and my client bought your product. I tried to find information how to forbid to create and upload folders to file manager, but I can't find. Help me please! Thanks!

**Comment by promatik** [comment, marked as answer]
Hi!
In order to control what gets done or not in file manager, you must use a middleware.

1) Go to `config\elfinder.php`, and add the `accessControl` key to your disk.
_Note that you can have multiple disks with different access control._

```php
'disks' => [
    'mydisk' => [
        'alias' => 'Storage',
        'accessControl' => 'App\Http\Middleware\ElfinderAccess::checkAccess',
        'visibility' => 'public',
    ],
],
```

2) Create your `app\Http\MiddlewareElfinderAccess.php` middelware.

```php
<?php
namespace App\Http\Middleware;

class ElfinderAccess
{
    public static function checkAccess($attr, $path, $data, $volume)
    {
        if (...) { // your rules
            return false;
        }

        return true;
    }
}
```

Regarding the rules you need, I'm not quite sure about it, but I think you'll need to check for the commands.

```php
if($attr === 'mkdir') {
    return false;
}
```

Since there isn't much information on the internet about it, probably the easiest is to `dd($attr)` your requests and start from there.
